server/src/modules/users/domain/value-objects/encrypted-secret.ts

```diff
@@ -0,0 +1,64 @@
+import bcrypt from 'bcrypt'
+
+import { Result } from '../../../../shared/core/result'
+import { ValueObject } from '../../../../shared/domain/value-object'
+import { UserValueObjectErrors } from './errors'
+
+export interface EncryptedSecretProps {
+  value: string
+  hashed: boolean
+}
+
+//The protected and public members of this base class can be inherited as needed
+export abstract class EncryptedSecret extends ValueObject<EncryptedSecretProps> {
+  protected static minLength: number = 64
+
+  public constructor(props: EncryptedSecretProps) {
+    super(props)
+  }
+
+  protected static isAppropriateLength(secret: string): boolean {
+    return secret.length >= this.minLength
+  }
+
+  get value(): string {
+    return this.props.value
+  }
+
+  public isAlreadyHashed(): boolean {
+    return this.props.hashed
+  }
```

Comment:
nit: This makes more sense to me:

```suggestion
  get isHashed(): boolean {
    return this.props.hashed
  }
```

It would be consistent with `get value` above